engine/apps/api/views/escalation_chain.py

```diff
@@ -11,10 +12,22 @@
 from apps.api.serializers.escalation_chain import EscalationChainListSerializer, EscalationChainSerializer
 from apps.auth_token.auth import PluginAuthentication
 from common.api_helpers.exceptions import BadRequest
+from common.api_helpers.filters import ByTeamModelFieldFilterMixin, ModelFieldFilterMixin, get_team_queryset
 from common.api_helpers.mixins import ListSerializerMixin, PublicPrimaryKeyMixin, TeamFilteringMixin
 from common.insight_log import EntityEvent, write_resource_insight_log
 
 
+class EscalationChainFilter(ByTeamModelFieldFilterMixin, ModelFieldFilterMixin, filters.FilterSet):
+    team = filters.ModelMultipleChoiceFilter(
+        field_name="team",
+        queryset=get_team_queryset,
+        to_field_name="public_primary_key",
+        null_label="noteam",
+        null_value="null",
+        method=ByTeamModelFieldFilterMixin.filter_model_field_with_multiple_values.__name__,
+    )
```

Comment:
what do you think about deduplicating this filter instance into a new class that subclasses `filters.ModelMultipleChoiceFilter`? This same filter logic is defined in four spots ([here](https://github.com/grafana/oncall/pull/1528/files#diff-17e2f72d153f97776a8ff0dd5edbbc415df4c5d9943f2555ab94221e098530d1R20-R27), [here](https://github.com/grafana/oncall/pull/1528/files#diff-6e0af191419ed818d441082ae583552cb8c1f18e459ff6a593c8cd2b6b7baa6dR63-R70), [here](https://github.com/grafana/oncall/pull/1528/files#diff-4bab66851ffc1e7845a537e83ee5624c99176b2a2c60bf141213919c95c9e6e7R38-R45), and [here](https://github.com/grafana/oncall/pull/1528/files#diff-4582f10ddb8985ec764d973c8de1e598a0b48bff6a2bcd539ba7b093c3dc17c3R98-R105))